src/game_state/player.rs

```diff
@@ -72,186 +49,172 @@ pub struct ReportMessage {
-pub struct PlayerLookMessage {
-    pub conn_id: Uuid,
-    pub new_angle: Angle,
+    pub new_position: Option<Position>,
+    pub new_angle: Option<Angle>,
 }
 
 pub fn start(receiver: Receiver<PlayerStateOperations>, messenger: Sender<MessengerOperations>) {
     let mut players = HashMap::<Uuid, Player>::new();
 
     while let Ok(msg) = receiver.recv() {
-        match msg {
-            PlayerStateOperations::New(msg) => {
-                let mut player = msg.player;
-                player.entity_id = players.len().try_into().expect("too many players");
-                let join_game = JoinGame {
-                    entity_id: player.entity_id,
-                    gamemode: 1,
-                    dimension: 0,
-                    difficulty: 0,
-                    max_players: 2,
-                    level_type: String::from("default"),
-                    reduced_debug_info: false,
-                };
-                send_packet!(messenger, msg.conn_id, Packet::JoinGame(join_game)).unwrap();
-                let player_pos_and_look = ClientboundPlayerPositionAndLook {
-                    x: player.position.x,
-                    y: player.position.y,
-                    z: player.position.z,
-                    yaw: 0.0,
-                    pitch: 0.0,
-                    flags: 0,
-                    teleport_id: 0,
-                };
-                send_packet!(
-                    messenger,
-                    msg.conn_id,
-                    Packet::ClientboundPlayerPositionAndLook(player_pos_and_look)
-                )
-                .unwrap();
-                send_packet!(
-                    messenger,
-                    msg.conn_id,
-                    Packet::ChunkData(ChunkData {
-                        chunk_x: 0,
-                        chunk_z: 0,
-                        full_chunk: true,
-                        primary_bit_mask: 1,
-                        size: 12291, //I just calculated the length of this hardcoded chunk section
-                        data: ChunkSection {
-                            bits_per_block: 14,
-                            data_array_length: 896,
-                            block_ids: Vec::new(),
-                            block_light: Vec::new(),
-                            sky_light: Vec::new(),
-                        },
-                        biomes: vec![127; 256],
-                        number_of_block_entities: 0,
-                    })
-                )
-                .unwrap();
-                players.insert(msg.conn_id, player);
-            }
-            PlayerStateOperations::Move(msg) => {
-                let player = players.get(&msg.conn_id).unwrap();
-                let mut player_clone = player.clone();
-                let position_delta =
-                    PositionDelta::new(player_clone.position, msg.new_position.clone());
+        handle_message(msg, &mut players, messenger.clone())
+    }
+}
+
+fn handle_message(
```

Comment:
Trying to reduce the amount of indentation used for readability purposes. I think this indirection is fine given these methods are right next to each other